module/VuFind/src/VuFind/ILS/Driver/Demo.php

```diff
@@ -1456,14 +1456,23 @@ public function getRenewDetails($checkOutDetails)
      * @param array  $data   An Array of item data
      * @param patron $patron An array of patron data
      *
-     * @return bool True if request is valid, false if not
+     * @return mixed An array of data on the request including
+     * whether or not it is valid and a status message. Alternatively a boolean
+     * true if request is valid, false if not.
      *
      * @SuppressWarnings(PHPMD.UnusedFormalParameter)
      */
     public function checkRequestIsValid($id, $data, $patron)
     {
         $this->checkIntermittentFailure();
-        return !$this->isFailing(__METHOD__, 10);
+        if ($this->isFailing(__METHOD__, 10)) {
+            return [
+                'valid' => false,
+                'status' => rand() % 3 == 0
+                    ? 'hold_items_available' : 'hold_error_blocked'
+            ];
+        }
+        return rand() % 2 == 0 ? ['valid' => true] : true;
```

Comment:
I think it might be better to just stick with using the new return format here, rather than randomizing the format. While it's true that this randomization might help catch a bug at some point, I think it would be somewhat hard to track down based on this, so it may be simpler just to leave it alone. If we want to avoid regressions, it would probably be more appropriate to figure out a way to do that with a test suite (not that I'm saying you need to do that right now).